src/routes/concepts/components/event-handlers.mdx

```diff
@@ -3,68 +3,209 @@ title: Event Handlers
+This means that event listeners are attached to the `document` element, and dispatched to the relevant elements as the event bubbles up.
 
 By keeping the number of event listeners to a minimum, events can be captured more effectively.
-This is especially useful when working with large lists of elements, such as in a table or list.
+This is especially useful when working with a large number of elements, such as in a table or list.
 
-Supported events such as `click`, `mouseover`, and `keydown` are just a few examples that are optimized in this way.
+Supported events such as `click`, `input` and `keydown` are just a few examples that are optimized in this way.
 To view the full list see the [references below](#delegated-events).
 
-If you need to attach an event listener to an element that is not supported by Solid's event delegation, you can use the [on:](/reference/jsx-attributes/on-and-oncapture) prefix followed by the event name, and assign it a function.
+If you need to attach an event listener to an element that is not supported by Solid's event delegation, such a custom event in a [Custom Element](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements), you can use the [`on:__`](/reference/jsx-attributes/on-and-oncapture) form.
 
 ```tsx
 <div on:customEvent={handleCustomEvent} />
 ```
 
-## Binding events
+### Event delegation considerations
+
+While delegated events provide some performance enhancements, it comes with some tradeoffs that should be considered.
+
+<Callout type="info" title="Design">
 
-In Solid, event handlers can be bound by passing an array to the event handler.
-The second item of the array is supplied as the handler's first argument.
+    Event delegation is designed for event propagation through the JSX Tree, rather than the DOM Tree.
+
+</Callout>
+
+This differs from the acquired expectations one may have over how events work and flow.
+
+Here is a list of peculiarities to keep in mind:
+
+1. Delegated event listeners are added to the `document` once per event type.
+   When the element/component that added it is removed, it's handler function will no longer be called, but the delegated event listener will still be present on the `document`.
+   Considering the following example, a `div` listening for `mousemove`.
+   `mousemove` will still be dispatched to the `document` even if the `div` is disposed.
 
 ```tsx
-const handler = (data, event) => {
-	console.log("Data:", data, "Event:", event);
-};
+<div onMouseMove={handleCustomEvent} />
+```
 
-<button onClick={[handler, "Hello!"]}>Click Me</button>;
+<Callout type="tip" title="Ocassional Events">
+
+    Consider using Native Events instead of Delegated Events for rare or ocassional events that happen on special circumstances and won't benefit from the performance improvements of event delegation.
```

Comment:
```suggestion
Rather than using delegated events for events that happen infrequently, **native events** are a better solution.
Since these events happen in specific circumstances, they do not benefit from the performance improvements of event delegation.
```